pkg/cache/azure_cache.go

```diff
@@ -64,8 +65,8 @@ func cacheKeyFunc(obj interface{}) (string, error) {
 
 // Resource operations
 type Resource interface {
-	Get(key string, crt AzureCacheReadType) (interface{}, error)
-	GetWithDeepCopy(key string, crt AzureCacheReadType) (interface{}, error)
+	Get(ctx context.Context, key string, crt AzureCacheReadType) (interface{}, error)
+	GetWithDeepCopy(ctx context.Context, key string, crt AzureCacheReadType) (interface{}, error)
 	Delete(key string) error
 	Set(key string, data interface{})
 	Update(key string, data interface{})
```

Comment:
Will you add ctx for these 3 operations?

Reply:
no. get function will send request to arm api. However, these three will only update local cache store. 
It will be easier if ctx is passed through when adding tracing support in cloud provider code base.

Reply:
I am thinking "program to interfaces, not implementations" though 😂

Reply:
I agree. Lets adopt context package step by step.